crates/wasmtime/src/runtime/component/concurrent.rs

```diff
@@ -3970,7 +3979,7 @@ pub struct ConcurrentState {
     /// out, poll it, then put it back to avoid any mutable aliasing hazards.
     futures: Mutex<Option<FuturesUnordered<HostTaskFuture>>>,
     /// The table of waitables, waitable sets, etc.
-    table: Table,
+    table: Mutex<ResourceTable>,
```

Comment:
Could this replace `Mutex` with `AlwaysMut` added in https://github.com/bytecodealliance/wasmtime/pull/11453?